dkg-gadget/src/testing.rs

```diff
@@ -0,0 +1,1210 @@
+		let (tx, mut rx) = dkg_mock_blockchain::transport::bind_transport::<TestBlock>(socket);
+		let tx0 = Arc::new(tokio::sync::Mutex::new(tx));
+		let tx1 = tx0.clone();
+
+		let this = TestBackend {
+			inner: TestBackendState {
+				finality_stream: Arc::new(MultiSubscribableStream::new("finality_stream")),
+				import_stream: Arc::new(MultiSubscribableStream::new("import_stream")),
+				api,
+				offchain_storage: Default::default(),
+				local_test_cases: Arc::new(Mutex::new(Default::default())),
+			},
+		};
+
+		let _this_for_dkg_listener = this.clone();
+		let logger0 = logger.clone();
+		let dkg_worker_listener = async move {
+			while let Some((trace_id, result)) = from_dkg_worker.recv().await {
+				logger0.info(format!(
+					"The client {peer_id:?} has finished test {trace_id:?}. Result: {result:?}"
+				));
+				let success = result.is_ok();
+				let packet = ProtocolPacket::ClientToBlockChain {
+					event: MockClientResponse { error: result.err(), success, trace_id },
+				};
+				tx0.lock().await.send(packet).await.unwrap();
+			}
+
+			panic!("DKG worker listener ended prematurely")
+		};
+
+		let this_for_orchestrator_rx = this.clone();
+		let orchestrator_coms = async move {
+			logger
+				.info(format!("Complete: orchestrator<=>DKG communications for peer {peer_id:?}"));
+			while let Some(packet) = rx.next().await {
+				match packet {
+					ProtocolPacket::InitialHandshake => {
+						// pong back the handshake response
+						tx1.lock()
+							.await
+							.send(ProtocolPacket::InitialHandshakeResponse { peer_id })
+							.await
+							.unwrap();
+					},
+					ProtocolPacket::BlockChainToClient { trace_id, event } => {
+						*latest_test_uuid.write() = Some(trace_id);
+
+						match event {
+							MockBlockChainEvent::FinalityNotification { notification } => {
+								this_for_orchestrator_rx
+									.inner
+									.local_test_cases
+									.lock()
+									.insert(trace_id, None);
+								this_for_orchestrator_rx.inner.finality_stream.send(notification);
+							},
+							MockBlockChainEvent::ImportNotification { notification } => {
+								this_for_orchestrator_rx
+									.inner
+									.local_test_cases
+									.lock()
+									.insert(trace_id, None);
+								this_for_orchestrator_rx.inner.import_stream.send(notification);
+							},
+							MockBlockChainEvent::TestCase { trace_id: _, test: _ } => {
+								unimplemented!()
+							},
+						}
+					},
+					ProtocolPacket::Halt => {
+						logger.info("Received HALT command from the orchestrator".to_string());
+						return
+					},
+
+					packet => {
+						panic!("Received unexpected packet: {packet:?}")
+					},
+				}
+			}
+
+			panic!("The connection to the MockBlockchain died")
+		};
+
+		tokio::task::spawn(dkg_worker_listener);
+		tokio::task::spawn(orchestrator_coms);
+
+		Ok(this)
+	}
+}
+
+// each stream may be called multiple times. As such, we need to keep track of each subscriber
```

Comment:
```suggestion
// Each stream may be called multiple times. As such, we need to keep track of each subscriber
```
Can you please capitalize the beginning of your docs? Also please provide more clear docs. What is a `MultiSubscribableStream`. Please improve the formatting and consistency with respect to other docs in the repo.